incydr/utils.py

```diff
@@ -1,40 +1,170 @@
 from __future__ import annotations
 
-from collections.abc import MutableMapping
 from csv import DictReader
 from csv import DictWriter
 from io import IOBase
 from itertools import chain
+from itertools import repeat
 from pathlib import Path
 from typing import Dict
 from typing import Generator
 from typing import Iterable
 from typing import List
 from typing import TextIO
 from typing import Type
-from typing import TypeVar
 from typing import Union
 
 from pydantic import BaseModel
 from pydantic import ValidationError
+from rich.console import ConsoleRenderable
+from rich.console import Group
+from rich.console import group
+from rich.panel import Panel
+from rich.text import Text
 
 
-Model = TypeVar("Model", bound=BaseModel)
+def model_field_getter(model: BaseModel, path: list[str]):
+    """
+    Traverse a pydantic model and its sub-models to retrieve both the value and `ModelField` data for a given attribute
+    path.
 
+    For example, given the following model hierarchy:
 
-# https://stackoverflow.com/questions/6027558/flatten-nested-dictionaries-compressing-keys
-def flatten(d, parent_key="", sep="."):
-    items = []
-    for k, v in d.items():
-        new_key = parent_key + sep + k if parent_key else k
-        if isinstance(v, MutableMapping):
-            items.extend(flatten(v, new_key, sep=sep).items())
+        class Child(BaseModel):
+            field_1: str = Field("value", extra_data=1)
+            field_2: int = Field(1)
+
+        class Parent(BaseModel):
+            field: str = Field("default")
+            child: Child
+
+
+    model_field_getter(Parent(child=Child()),
```

Comment:
is there supposed to be more to this example? or am I just confused

Reply:
Nope, I failed to complete that. updated. 

Also, I think this function could be named better, open to suggestions.

Reply:
Hm i wonder if the name seems weird because it's using the noun `getter` instead of the usual `get` verb?  So maybe something like `get_model_field()` or `get_field_value()` ?